experimental/packages/opentelemetry-sdk-metrics-base/src/state/MetricCollector.ts

```diff
@@ -32,12 +32,12 @@ export class MetricCollector implements MetricProducer {
     this.aggregatorTemporality = this._metricReader.getPreferredAggregationTemporality();
   }
 
-  async collect(): Promise<MetricData[]> {
+  async collect(): Promise<MetricsData> {
     const collectionTime = hrTime();
     const results = await Promise.all(this._sharedState.meters
       .map(meter => meter.collect(this, collectionTime)));
 
-    return results.reduce((cumulation, current) => cumulation.concat(current), []);
+    return results.reduce((cumulation, current) => cumulation.merge(current), new MetricsData());
```

Comment:
When the meter returns a plain interface of `InstrumentationLibraryMetricsData`, it seems we don't need a `MetricsData.merge` here and exporters don't need `MetricsData.resourceMetrics` to transform maps of data to plain structures?

i.e. a `ResourceMetrics` could be constructed with an array of `InstrumentationLibraryMetricsData` and returned.

Reply:
Thanks! This made it possible to get rid of `MetricsData`